src/naive/date/mod.rs

```diff
@@ -1140,7 +1140,8 @@ impl NaiveDate {
         let (year2_div_400, year2_mod_400) = div_mod_floor(year2, 400);
         let cycle1 = yo_to_cycle(year1_mod_400 as u32, self.ordinal()) as i64;
         let cycle2 = yo_to_cycle(year2_mod_400 as u32, rhs.ordinal()) as i64;
-        TimeDelta::days((year1_div_400 as i64 - year2_div_400 as i64) * 146_097 + (cycle1 - cycle2))
+        let days = (year1_div_400 as i64 - year2_div_400 as i64) * 146_097 + (cycle1 - cycle2);
+        expect!(TimeDelta::try_days(days), "range of NaiveDate is MUCH less than TimeDelta")
```

Comment:
This error message is a bit too opinionated for my taste, can we just say it is out of range?

Reply:
It is const but otherwise functionally the same as
```rust
TimeDelta::try_days(days).unwrap() // range of NaiveDate is MUCH less than TimeDelta
```

The doc comment a little above says:
```
    /// This does not overflow or underflow at all,
    /// as all possible output fits in the range of `TimeDelta`.
```

The range of `TimeDelta` is ca. 585 million years, the range of `NaiveDate` ca. 525.000 years.

I can change it to `"always in rage"`.

Reply:
That works for me.